welly/las.py

```diff
@@ -0,0 +1,366 @@
+                                - every column represents a data variable. Column name is the data variable mnemonic.
+        header (pd.DataFrame): where:
+                                - every row represents a line read from LAS file and columns.
+                                - column 1-5 - directly parsed from the HeaderItem.__dict__:
+                                    'original_mnemonic' - original mmnemonic
+                                    'mnemonic' - mnemonic
+                                    'unit' - unit
+                                    'value' - value
+                                    'descr' - description
+                                - column 6 - is added as an identifier to track of sections:
+                                    'section' (str) - Section name the line from the LAS file belongs to.
+
+        Example of how returns of a 'data' and 'header' object would look:
+
+        data = pd.DataFrame({
+            'DEPT': [100.0, 101.0, 102.0],
+            'GR': [80.0, 85.0, 82.0],
+            'DEN': [2.10, 2.15, 2.20]
+        })
+
+        header = pd.DataFrame({
+            'original_mnemonic': ['VERS', 'WRAP', 'STRT', 'STOP', 'STEP', 'DEPT', 'GR', 'DEN', ''],
+            'mnemonic': ['VERS', 'WRAP', 'STRT', 'STOP', 'STEP', 'DEPT', 'GR', 'DEN', ''],
+            'unit': ['', '', 'M', 'M', 'M', 'M', 'GAPI', 'g/cm3', '']
+            'value': [2.0, 'NO', 100.0, 102.0, 1.0, '', '', '', '']
+            'descr': ['Version 2.0', 'One line per depth step', '', '', '', 'DEPTH', 'Gamma Ray', 'Density', 'Comment']
+            'section': ['Version', 'Version', 'Well', 'Well', 'Well', 'Curves', 'Curves', 'Curves', 'Other']
+        })
+    """
+    # read las file
+    las = lasio.read(file_ref, **kwargs)
+
+    datasets = from_lasio(las)
+
+    return datasets
+
+
+def from_lasio(las):
+    """
+
+    """
+    # get las format version
+    version = float(las.version[0].value)
+
+    if version == 2.0 or version == 1.2:
+        datasets = from_las_2_or_older(las)
+    else:
+        try:
+            datasets = from_las_2_or_older(las)
+            warnings.warn(f"Warning, LAS version {version} not yet supported. "
+                          "Attempting to use LAS 1.2 and 2.0 parsing logic "
+                          "for LAS 3.0.")
+        except Exception:
+            raise NotImplementedError(
+                f"LAS version {version} not yet supported")
+
+    return datasets
+
+
+def from_las_2_or_older(las):
+    """
+    Parse lasio.LASFile (LAS version 1.2 & 2.0) to 2 pd.DataFrames that form 1
+    datasets entry as a tuple.
+
+    For LAS versions 1.2 & 2.0, 1 LAS file translates to 'Curves' 1 data set.
+
+    Args:
+        las (lasio.LASFile): LASFile constructed through lasio.read()
+
+    Returns:
+        datasets = {'Curves': (data, header)}
+
+        data   (pd.DataFrame): df where:
+                                - indexed by order of vertical occurrence in las file.
```

Comment:
I don't know what this means: indexed by order of vertical occurrence in las file, vertical occurrence of what?